errbot/backends/slack.py

```diff
@@ -565,6 +566,25 @@ def send_message(self, mess):
                 "to %s: %s" % (to_humanreadable, mess.body)
             )
 
+    def _slack_upload(self, stream):
+        """Perform upload defined in a stream."""
+        stream.accept()
```

Comment:
The final step in fixing it for me was removing the debug statements inside the upload statement. Having them in before and after the `api_call` caused the upload to fail silently, removing them made the upload succeed.

I've wrapped the method in a try/except block to print out a failure message on upload, as requested.